### PR TITLE
Consolidate macos pip dependencies

### PR DESCRIPTION
**Pull request by huydhn**
After conda, consolidating all macos pip dependencies to cache every dependencies that macos CI needs. Two small issues are found along the way in `_mac-test-mps` workflow:

* It didn't have `Install macOS homebrew dependencies` to install libomp like the regular `_mac-test` workflow
* It didn't install `scipy`, thus silently skipping some `signal.windows` tests

Both are fixed in this PR

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
This PR makes me sad - instead of consolidating it creates two almost identical lists. 
Can we use environment marker instead to highlight the differences between M1 and Intel?
Something like `; platform_machine == 'arm64'`?

File: .github/requirements/pip-requirements-macOS [hunk 1]
```diff
@@ -0,0 +1,21 @@
+boto3==1.19.12
+hypothesis==6.56.4
+expecttest==0.1.3
+librosa>=0.6.2
+mpmath==1.2.1
+networkx==2.8.7
+numba==0.56.0; platform_machine == "arm64"
+numba<=0.49.1; platform_machine == "x86_64"
```
Comment (malfet) [inline .github/requirements/pip-requirements-macOS:8]: Please add comments with some explanations, why this discrepancy exists
```suggestion
# Use numba-0.49.1 or older on Intel Macs, but 0.56.0 on M1 machines, as older numba is not available 
numba==0.56.0; platform_machine == "arm64"
numba<=0.49.1; platform_machine != "arm64"
```

File: .github/requirements/pip-requirements-macOS [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+opt-einsum>=3.3
+psutil==5.9.1
+pynvml==11.4.1
+pygments==2.12.0
+pytest==7.2.0
+pytest-xdist==3.0.2
+pytest-rerunfailures==10.2
+pytest-shard==0.1.2
+scipy==1.9.0; platform_machine == "arm64"
+scipy==1.6.3; platform_machine == "x86_64"
```
Comment (malfet) [inline .github/requirements/pip-requirements-macOS:18]: Perhaps as followup: why can't we run both on 1.9.0?

**Review by malfet — APPROVED** [review]
Looks good to me, though please add .txt extension to [pip-requirements-macOS](https://github.com/pytorch/pytorch/pull/88071/files#diff-e5febc2626b3a1d99249bb25dbcd5e6859f736f616ee18d06022a279890429f8)

**Comment by malfet** [comment]
Also, what those requirements have to do with GitHub? @janeyx99 tried at some point to move configuration files specific to CI but agnostic to specific runner infrastructure to `.ci` folder, perhaps we should continue with the tradition

File: .github/requirements/pip-requirements-macOS (hunk 2 above)
Comment (huydhn) [reply .github/requirements/pip-requirements-macOS:19]: This is copied from the shell script, so let me try if this is also work with 1.9.0

**Comment by huydhn** [comment]
> Also, what those requirements have to do with GitHub? @janeyx99 tried at some point to move configuration files specific to CI but agnostic to specific runner infrastructure to `.ci` folder, perhaps we should continue with the tradition

Noted. Once this is done, I will push another PR to move all the requirements to a new `.ci` folder. That should be a simple change I guess.  There is no particular reason why they need to be under .github directory

**Pull request closed without merging**